test/e2e/framework/networking_utils.go

```diff
@@ -220,16 +220,16 @@ func (config *NetworkingTestConfig) DialFromContainer(protocol, containerIP, tar
 	Failf("Failed to find expected endpoints:\nTries %d\nCommand %v\nretrieved %v\nexpected %v\n", maxTries, cmd, eps, expectedEps)
 }
 
-func (config *NetworkingTestConfig) GetEndpointsFromTestContainer(protocol, targetIP string, targetPort, tries int) (sets.String, error) {
-	return config.GetEndpointsFromContainer(protocol, config.TestContainerPod.Status.PodIP, targetIP, TestContainerHttpPort, targetPort, tries)
+func (config *NetworkingTestConfig) GetDiffEndpointsFromTestContainer(protocol, targetIP string, targetPort, tries int) (sets.String, error) {
+	return config.GetDiffEndpointsFromContiner(protocol, config.TestContainerPod.Status.PodIP, targetIP, TestContainerHttpPort, targetPort, tries)
 }
 
-// GetEndpointsFromContainer executes a curl via kubectl exec in a test container,
+// GetDiffEndpointsFromContainer executes a curl via kubectl exec in a test container,
 // which might then translate to a tcp or udp request based on the protocol argument
-// in the url.
-// - tries is the maximum number of curl attempts. If this many attempts pass and
+// in the url. It returns all different endpoints from multiple retries.
+// - tries is the number of curl attempts. If this many attempts pass and
 //   we don't see any endpoints, the test fails.
-func (config *NetworkingTestConfig) GetEndpointsFromContainer(protocol, containerIP, targetIP string, containerHttpPort, targetPort, tries int) (sets.String, error) {
+func (config *NetworkingTestConfig) GetDiffEndpointsFromContiner(protocol, containerIP, targetIP string, containerHttpPort, targetPort, tries int) (sets.String, error) {
```

Comment:
Why the rename? Diff to me implies comparing or a delta between an old and new.  I don't think it's doing that here?  As a name, `GetEndpointsFromContainer` already tells me it's getting all of the container's endpoints?

Reply:
I guess it looks like the original impl of this was buggy and returned too soon? Still think the name fits the intent here

Reply:
+1, `GetEndpointsFromContainer` seems to be a proper name.